core/state/pruner/pruner.go

```diff
@@ -265,7 +265,7 @@ func (p *Pruner) Prune(root common.Hash) error {
 	// Ensure the root is really present. The weak assumption
 	// is the presence of root can indicate the presence of the
 	// entire trie.
-	if blob := rawdb.ReadTrieNode(p.db, root); len(blob) == 0 {
+	if blob := rawdb.ReadLegacyTrieNode(p.db, root); len(blob) == 0 {
```

Comment:
-> `if !rawdb.HasLegacyTrieNode(p.db, root)` Could we change to this logic ? hm no need for reading data here right ?

Reply:
You're right. Updated.